Kernel/Library/string.cpp

```diff
@@ -138,11 +138,27 @@ void* memmove(void* destptr, const void* srcptr, size_t size)
     return destptr;
 }
 
-void* memcpy(void* dstptr, const void* srcptr, size_t size)
+void* memcpy(void* dst, const void* src, size_t size)
 {
-    unsigned char* dst = (unsigned char*)dstptr;
-    const unsigned char* src = (const unsigned char*)srcptr;
-    for (size_t i = 0; i < size; i++)
-        dst[i] = src[i];
-    return dstptr;
+    uint8_t *bDst;
+    uint8_t *bSrc;
+    uint32_t *lDst = (uint32_t*)dst;
```

Comment:
Why not use `size_t`'s here?

Reply:
Because that's how liballoc had it. You're right, probably should be size_t's though.

Reply:
Yeah and I have no idea if they need to unaligned start/destination addresses, but that might be something to account for seeing as this is indented to be a generic `memcpy` replacement.